---
title: "9.Transcriptome alignment assessment"
author: "CTLai"
date: "March 31, 2018"
---

### Transcriptome alignment assessment

In the previous video we calculated expression by probabilistically aligning reads with the RSEM software. And we have provided for you two results files:

genes.results (.5 Mb)

isoforms.results (1 Mb)

We will investigate the values for various columns in these tables. For reference, the documentation for these file formats is here.

First, read these into an R session with the following lines:

```{r}
genes <- read.table("SRR1039508.genes.results", header=TRUE)
isoforms <- read.table("SRR1039508.isoforms.results", header=TRUE)
```

Both 'genes' and 'isoforms' have a column FPKM. 

Using split() and sapply(), confirm that the FPKM column in 'genes' is the sum of the FPKM column in 'isoforms':

```{r}
fpkm.per.gene <- split(isoforms$FPKM, isoforms$gene_id)
head(sapply(fpkm.per.gene, sum))
head(genes$FPKM)
```
	
Or equivalently using dplyr:

#if not yet installed: install.packages("dplyr")
```{r}
library(dplyr)
isoforms %>% group_by(gene_id) %>% summarize(sum = sum(FPKM))  
```

Make a histogram of the FPKM column in 'genes'. Make a histogram after transforming by: log10(x+1).

Make new tables, removing the genes with FPKM 0:

```{r}
genes2 <- genes[genes$FPKM > 0,]
genes2$gene_id <- droplevels(genes2$gene_id)
isoforms2 <- isoforms[isoforms$gene_id %in% genes2$gene_id,]
isoforms2$gene_id <- droplevels(isoforms2$gene_id)
```

We will continue below using these new tables.

We perform a check that the gene_id column is genes2 is equal to the levels of the gene_id factor in isoforms2:

```{r}
stopifnot(all(genes2$gene_id == levels(isoforms2$gene_id)))
```

If this line runs without error, then we know the check passed.

With 'genes2', make a plot of the effective_length (x-axis) and the expected_count (y-axis), with both axes on the log scale: log="xy".

```{r}
plot(genes2$effective_length,genes2$expected_count,log="xy")
```

#### Transcript align Q1

Make a histogram of the FPKM in genes2. Make a histogram after transforming with: log10(x).

What is the median FPKM of the genes with positive FPKM (i.e. the genes2 table)?

##### ANS: 21.94


```{r}
hist(genes2$FPKM,breaks=50)
hist(log10(genes2$FPKM+1),breaks=50)
summary(genes2$FPKM)
```

#### Transcript align Q2 

With 'isoforms2', calculate the maximum IsoPct for each gene, 'max.iso'. Make at a histogram of 'max.iso'.

For what ratio of genes is the maximum IsoPct higher than 95? (Enter a number between 0 and 1.)

##### ANS: 0.603

```{r}
library(dplyr)
res<-isoforms2 %>% group_by(gene_id) %>% summarize(max.iso = max(IsoPct))
hist(res$max.iso)
mean(res$max.iso>95)
```

```{r}
isoforms2 %>% group_by(gene_id) %>% summarize(max.iso = max(IsoPct)) %>% summarize(answer = mean(max.iso > 95))
```

#### Transcript align Q3 

Is there a relationship between total expression and the maximum IsoPct per gene?

Make a plot of 'max.iso' on the x-axis and genes2$FPKM on the y-axis (with log="y").

```{r}
plot(res$max.iso,genes2$FPKM,log="y")
```

Sometimes it is difficult to see relationships when many points stack at one value. Try a boxplot:

```{r}
max.iso = sapply(split(isoforms2$IsoPct, isoforms2$gene_id), max)
boxplot(split(log10(genes2$FPKM), cut(max.iso, 5)), xlab="max.iso", ylab="log10 FPKM")
```

Which group has the lowest median FPKM? Genes with 'max.iso' in the range:

1. 12.5 - 30.1

2. 30.1 - 47.6  

3. 47.6 - 65  

4. 65 - 82.5  

5. 82.5 - 100

##### ANS: 5
