---
title: "3.Transcript quantification assessment"
author: "CTLai"
date: "March 31, 2018"
output: html_document
---

###Transcript quantification assessment

In the previous video, we showed the model for quantification of each of three possible transcripts (also called "isoforms") of a gene. Here we will try to find the maximum likelihood estimate of for a given set of counts and lengths.

$E \left( \begin{array}{c} Y_1 / w l_1 \\ Y_2 / w l_2 \\ Y_3 / w l_3 \\ Y_{1,2} / w l_{1,2} \\ Y_{2,3} / w l_{2,3} \end{array} \right) = \left( \begin{array}{ccc} 1 & 1 & 0 \\ 1 & 1 & 1 \\ 0 & 1 & 1 \\ 1 & 1 & 0 \\ 0 & 1 & 1 \end{array} \right) \left( \begin{array}{c} \theta _1 \\ \theta _2 \\ \theta _3 \end{array} \right)$

For this assessment we will consider only a single gene for simplicity.

The l's give the lengths for the regions which contribute to the exon counts for 1, 2 and 3 and to the junction counts for {1,2} and for {2,3}. We will assume these counts are disjoint: so a read counts for an exon only if it does not cross a junction. This is a simplified model, which allows independence of the counts.

Suppose the following lengths:

```{r}
l1 = 100
l2 = 200
l3 = 300
l12 = 100
l23 = 100
```

We can write the lengths vector in R:

```{r}
lengths = c(100,200,300,100,100)
```

The matrix which identifies transcripts with exons and junctions:

```{r}
mat = cbind(c(1,1,0,1,0),c(1,1,1,1,1),c(0,1,1,0,1))
```

The length of the transcripts is then: lengths %*% mat 

Suppose we align 1000 reads to this gene. So 

```{r}
w = 1000
```

Suppose we observe the following read counts for the exons and for the two junctions:

```{r}
counts = c(125,350,300,125,100)
````

Given the formula above, and the assumption of uniform read distribution and Poisson counts, we can get a rough estimate of theta by just solving the linear system above. The estimate is only a rough one, because we will not use the probabilistic model in the previous video. We will walk through the estimation step-by-step below.

First try a guess at theta:

```{r}
theta.hat = c(1, 2, 3) / 10000
```

We can see with this guess, our counts are too low, and not properly apportioned to the exons and junctions:

```{r}
mat %*% theta.hat * lengths * w
```

We can roughly estimate theta by solving the linear system of equations above:

```{r}
LHS = counts/(lengths * w)
lm.fit(mat, LHS)$coefficients
```

We note we are only roughly estimating theta, because we are not using the likelihood from our probabilitistic model for counts, but simply minizing the squared error between the left-hand side (LHS) and the right-hand side of the equation (recall the linear models covered in PH525.2x).

####Transcript quant. Q1 

What is the estimate of theta using our rough estimator, for the first transcript (the transcript with exon 1 and exon 2)?

#####ANS: 0.00075

```{r}
mat %*% theta.hat * lengths * w
lm.fit(mat, LHS)$coefficients[1]
```

####Transcript quant. Q2

What would be the rough estimate of theta for the first transcript if the counts for exons and junctions were 60,320,420,60,140?

#####ANS: 0.0002

```{r}
counts2=c(60,320,420,60,140)
LHS2 = counts2/(lengths * w)
theta.hat2=lm.fit(mat, LHS2)$coefficients
theta.hat2[1]
```

