---
title: "6.Genome alignment assessment"
author: "CTLai"
date: "March 31, 2018"
---

### Genome alignment assessment

The head of the Aligned.out.sam file is pasted below:

    @PG     ID:STAR PN:STAR VN:STAR_2.3.1z_r395     CL:STAR   --runThreadN 12   --genomeDir GRCh38.79.chrom1   --readFilesIn fastq/SRR1039508_1.fastq   fastq/SRR1039508_2.fastq    cl:STAR --runThreadN 12 --genomeDir GRCh38.79.chrom1 --readFilesIn fastq/SRR1039508_1.fastq fastq/SRR1039508_2.fastq
    @SQ     SN:1    LN:248956422
    SRR1039508.7    163     1       96446974        255     63M     =       96446997        86      CGTAGATTCGGGCAAGTCCACCACTACTGGCCATCTGATCTATAAATGCGGTGGCATCGACAA HJHIJJJJJJJJJJJJJJJIJJJJJJJJIJJJJJJJJJIIJJJJJJJJJJJEHHFFFFDEDDD NH:i:1  HI:i:1  AS:i:110        nM:i:7
    SRR1039508.7    83      1       96446997        255     63M     =       96446974        -86     CTACTGGCCATCTGATCTATAAATGCGGTGGCATCGACAAAAGAACCATTGAAAAATTTGAGA FHHHHHJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJIJJJJJJJJJJJJJJJJJH NH:i:1  HI:i:1  AS:i:110        nM:i:7
    SRR1039508.16   163     1       243426450       255     63M     =       243426495       108     AAGAAAAAAGGTATACATATGATAAATTGGGAAAGTTACAGAGAAGAAATGAAGAATTGGAGG HJJJJJJJJJJAGIJJJIJJJJJJJJJJIJJIIJJHIJFHJIJJIJIJJJJJJJJIJJHHHHG NH:i:1  HI:i:1  AS:i:124        nM:i:0
    SRR1039508.16   83      1       243426495       255     63M     =       243426450       -108    GAAATGAAGAATTGGAGGAACAGTGTGTCCAGCATGGGAGAGTACATGAGACGATGAAGCAAA IJJIGJJJJJIIIJJJJJIIJJJJJJJIJJJJJIGJJIJJJJJIJJJJIJJJJJJJJJJJJJH NH:i:1  HI:i:1  AS:i:124        nM:i:0
    

The SAM format and its compressed version, BAM, contains information about the alignment of sequences to a reference genome.

Section 1.3 describes the header of the SAM file.

#### Genome align Q1

What is the length of reference sequence 1 according to our SAM file?

##### ANS: 248956422

#### Genome align Q2

In section 1.4 on the alignments, the different fields are described. The CIGAR string tells us how the read aligned to the genome, using numbers and letters, M for match, I for insertion, D for deletion, N for skipped bases on the reference genome.
What is the CIGAR string for all of the two pairs of reads we are shown above? (The answer includes numbers and a letter)

##### ANS: 63M

In section 1.4, we see that the second column is a bitwise FLAG, an integer which encodes possible YES/NO answers to the choices below.

The bitwise FLAG is most easily decoded at this link, Explain SAM Flags, for the PICARD software.

#### Genome align Q3 

Type the bitwise FLAG of the top read above into the SAM flag explainer. Which choices are true about this read?

1.read paired

2.read mapped in proper pair

3.read unmapped

4.mate unmapped

5.read reverse strand

6.mate reverse strand

7.first in pair

8.second in pair

9.not primary alignment

10.read fails platform/vendor quality checks

11.read is PCR or optical duplicate

12.supplementary alignment

##### ANS: 1,2,6,8

